File: c++/5430.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main() {
  freopen("example.txt", "r", stdin);
  int t;
  cin >> t;

  for (int tc = 0; tc < t; tc++) {
    string p, s;
    int n;
    cin >> p >> n >> s;

    vector<string> v;
    string ss = "";
    for (int i = 1; i < s.length(); i++) {
      if ((s[i] == ']' || s[i] == ',') && ss.length() != 0) {
        v.push_back(ss);
        ss = "";
      } else {
        ss += s[i];
      }
    }

    bool isFlip = false;
    int start = 0;
    int end = n;

    for (int i = 0; i < p.length(); i++) {
      if (p[i] == 'R')
        isFlip = !isFlip;
      else
        isFlip ? end-- : start++;
    }

    if (start > end)
      cout << "error" << '\n';
    else {
      cout << '[';
      for (int i = 0; i < end - start; i++) {
        cout << (isFlip ? v[end - 1 - i] : v[start + i]);
        if (i != end - start - 1) cout << ',';
      }
      cout << "]\n";
    }
  }

  return 0;
}

File: js/5430.js
const fs = require('fs');

const inputs = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const t = +inputs[0];
const answer = [];

for (let tc = 1; tc < t * 3; tc += 3) {
  const p = inputs[tc].split('');
  const n = +inputs[tc + 1];
  const arr = JSON.parse(inputs[tc + 2]);

  let isFlip = false;
  let start = 0;
  let end = n;
  p.forEach((command) => {
    if (command === 'R') isFlip = !isFlip;
    else {
      if (isFlip) end--;
      else start++;
    }
  });

  if (start > end) {
    answer.push('error');
  } else {
    const temp = [];
    for (let i = 0; i < end - start; i++) {
      temp.push(isFlip ? arr[end - 1 - i] : arr[start + i]);
    }
    answer.push(`[${temp.toString()}]`);
  }
}

console.log(answer.join('\n'));
